client/tx/search.go

```diff
@@ -165,17 +174,47 @@ func SearchTxRequestHandlerFn(cliCtx context.CLIContext, cdc *codec.Codec) http.
 		}
 
 		for key, values := range r.Form {
+			if key == "page" || key == "limit" {
+				continue
+			}
 			value, err := url.QueryUnescape(values[0])
 			if err != nil {
 				utils.WriteErrorResponse(w, http.StatusBadRequest, sdk.AppendMsgToErr("could not decode query value", err.Error()))
 				return
 			}
 
-			tag := fmt.Sprintf("%s='%s'", key, value)
+			var tag string
+			if key == types.TxHeightKey {
+				tag = fmt.Sprintf("%s=%s", key, value)
+			} else {
+				tag = fmt.Sprintf("%s='%s'", key, value)
+			}
 			tags = append(tags, tag)
 		}
 
-		txs, err = SearchTxs(cliCtx, cdc, tags)
+		pageStr := r.FormValue("page")
+		if pageStr == "" {
+			pageStr = "0"
+		}
+		page, err := strconv.Atoi(pageStr)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
```

Comment:
we should also use `utils.WriteErrorResponse()` here

Reply:
Agreed